scripts/create-cohort-project.mjs

```diff
@@ -190,8 +190,11 @@ const initRepo = async (repoDir, opts) => {
 
   console.log('Initializing repo...');
   await exec('git init', { cwd: repoDir });
-  await exec('git add .', { cwd: repoDir });
-  await exec('git commit -m "chore(init): Adds project files from curriculum"', { cwd: repoDir });
+  const { stdout: hasChanges } = await exec('git status --porcelain', { cwd: repoDir });
+  if (!!hasChanges) {
+    await exec('git add .', { cwd: repoDir });
+    await exec('git commit -m "chore(init): Adds project files from curriculum"', { cwd: repoDir });
+  }
```

Comment:
Aqui pasa que - si repite el script y sobreescribe el dir de proyecto, el commit [tira el error](https://github.com/Laboratoria/bootcamp/pull/1348#issuecomment-1481911754) porque no tiene nada para . 
No se si este es el comportamiento que queremos tener - pensaria que queremos arreglarlo.

Reply:
hola @unjust! estoy teniendo problemas para reproducir este error antes de integrar tu PR, a mi no me tira error, simplemente me tira el prompt de la función `ensureRepoDir`:

> Dir /home/alpi/Documents/TEST-cipher already exists.. use it anyway? [y/N]:

si le doy `N` arroja el error con mensaje `Aborting` como está descrito en la función.

si le doy `y` simplemente continúa ejecutando `mkdirp(repoDir)` e inicializa el repo sin problemas y me dice `Done` al final.

Al preguntarle el código de exit del proceso me dice `0`, por lo que funciona todo bien! :thinking: debe ser _otra_ cosa lo que te está dando error, pero para estar seguro lo probé en dos compus distintos con OS de diferentes versiones y en ambos funciona correctamente!

Reply:
ok posiblemente no es una problema en general - y puedo revertir el commit. Pero los pasos para duplicar para mi son:

En la rama main:

1. `GITHUB_TOKEN=[YOUR_TOKEN] npm run create-cohort-project projects/04-notes ../repos/cohorts DEV888`

2. `You are about to copy all files from projects/04-notes to ../repos/cohorts/DEV888-notes
Are you sure you want to continue? [Y/n]: Y`
3. `Would you like to create a repository on GitHub and push changes? [Y/n]: n`
4. `Done`
5. `GITHUB_TOKEN=[YOUR_TOKEN] npm run create-cohort-project projects/04-notes ../repos/cohorts DEV888`
6. `Dir ../repos/cohorts/DEV888-notes already exists.. use it anyway? [y/N]: y`
7. `You are about to copy all files from projects/04-notes to ../repos/cohorts/DEV888-notes
Are you sure you want to continue? [Y/n]: Y`

```
Copying files...
Initializing repo...
Error: Command failed: git commit -m "chore(init): Adds project files from curriculum"

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:320:12) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'git commit -m "chore(init): Adds project files from curriculum"',
  stdout: 'On branch main\nnothing to commit, working tree clean\n',
  stderr: ''
}
```